doc/user_guide/io.rst

```diff
@@ -1047,39 +1047,50 @@ Extra loading arguments
   The underlying method of downsampling is unchangeable: sum. Differently than
   ``block_reduce`` from skimage.measure it is memory efficient (does not creates
   intermediate arrays, works inplace).
-- ``cutoff_at_kV`` : if set (can be int or float >= 0) can be used either to crop
-  or enlarge energy (or channels) range at max values (default None).
+- ``cutoff_at_kV`` : if set (can be None, int, float (kV), one of 'last_non-zero'
+  or 'em_hv') can be used either to crop or enlarge energy (or number of
+  channels) range at max values. It can be used to conserve memory or enlarge
+  the range if needed to mach the size of other file. Default value is None
+  (which does not influence size). Numerical values should be in kV.
+  "last_non-zero" truncates to the last non zero channel (this option
+  should not be used for stacks, as low beam current EDS can have different
+  last non zero channel per slice). "em_hv" truncates channels to SEM/TEM
+  acceleration voltage or energy at last channel, depending which is smaller.
```

Comment:
For `"em_hv"`, we would expect (from the name ) to consider only the acceleration voltage and not the last non zero channel. Maybe it would be good to have an `"auto"` option:
-  `"em_hv"` cut at the acceleration voltage
- `"auto"` cut at the minimum of acceleration voltage and last non zero energy channel

Reply:
last channel != last non-zero channel

Reply:
to last nonzero there is other option `last_non-zero`

Reply:
`auto` is really not good as it is not clearvwhat to expect

Reply:
Yes, sorry, I misread. Why do you need to consider the energy at last channel?

Reply:
case1: hv 25kV; channels scales up to 20kV; last non-zero channel coresponds to 20kV; `em_hv` cuts at 20kV; `last_non-zero` same.
case2: hv 15kV; channels scales up to 20kV; last non-zero channel (very light material) coresponds to 14.65kV; em_hv cuts at 15kV; `last_non-zero` at 14.65kV.

Reply:
in case of detailed single large hypermap last_non-zero will be most optimal option.

Reply:
and with esprit 2 64bit edition limit of 1024x768 size is lifted.

Reply:
also `last_non-zero` is safe auto option for xrf; where `em_hv` not.

Reply:
What I find inconsistent is that in case1 with `em_hv`, you end up with a value different than the acceleration voltage instead of extending the range of energy as it would do if you specify a value. If the energy range extension wouldn't be implemented when passing float value, I wouldn't find it "inconsistent". This is why I suggested to add the "auto" which will automatically choose the minimum between the two values.

This is not a big deal, but I think that adding the "auto" would be slightly more clearer.

Reply:
so probably I should switch `em_hv` back to `least_constraint` or `auto`. I forgot that I also included checing for no or 0.0 hv and in case of uXRF that would fall back to full channel count. So I guess then `auto` is really ok...

Reply:
now I am tempted to rename `last_non-zero` into `zealous` (like Gimp's zealous crop)